### PR TITLE
Assignment-2

### PR DESCRIPTION
**Pull request by Tom-Rohr**
New PR created from updated branch to avoid merge conflicts. Assignment-2 of my fork contains the full commit history for this assignment.

Link: https://polite-forest-0abe4e51e.3.azurestaticapps.net/

### CODE REVIEW

File: wordle-web/src/components/NavBar.vue [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        <v-btn
+          color="success"
+          size="medium"
+          class="ml-3 mt-1"
+          icon="mdi-cog-outline"
+          @click="dialog = true"
+        ></v-btn>
+        <v-dialog v-model="dialog" width="40%">
+          <v-card>
+            <!-- -----------------------------FOR TESTING CUSTOM THEMES------------------------------>
```
Comment (VForev) [inline wordle-web/src/components/NavBar.vue:39]: This color theme preview is super cool and useful for the dev. I personally was having the hardest time testing this when I couldn't see the colors.

**Review by VForev — COMMENTED** [review]
Create a custom component to show available words ✅
Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ✅
Create unit tests for the validWords method ✅
Display the number of valid words based on the entered letters ✅
When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
When a word is clicked, the word should be entered as the guess but not submitted ✅
Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅
Convert the sidebar to an App Bar ✅
Name of the app on the left with an icon ✅
Choose an icon for the app (from Material Design or somewhere on the web) ✅
Clicking the name and icon of the app take you to the home page ✅
Hamburger menu on the right ✅
Menu option for an About page ✅
Create an About page that has a few sentences about this class project and how great Meg is ✅
Make sure this has a back button ✅
Add a settings dialog ✅
Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
Option for dark and light mode ✅
Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
The above two items must be implemented with built in Veutify features ✅
Style the main game page ✅
Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
Add gradient to guesses blocks and keys (with CSS) ✅
Add drop shadow to keys (No CSS) ✅
Make the app responsive (No CSS) ✅
Include a link to your Azure App in your PR comments.
Extra Credit
Add an appropriate sound when the UI letter buttons are clicked (5 points)
Add creative styling following good UI/UX practices
Look at products by Google or some of the vuetify components for good UX/UI ideas
https://m2.material.io/design/guidelines-overview

**Review by alowe4 — COMMENTED** [review]
Create a custom component to show available words ✅

    Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ✅
    Create unit tests for the validWords method ✅
    Display the number of valid words based on the entered letters ✅
    When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
    When a word is clicked, the word should be entered as the guess but not submitted ✅
    Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

    Convert the sidebar to an App Bar ✅

    Name of the app on the left with an icon ✅
    Choose an icon for the app (from Material Design or somewhere on the web) ✅
    Clicking the name and icon of the app take you to the home page ✅
    Hamburger menu on the right ✅
    Menu option for an About page ✅
        Create an About page that has a few sentences about this class project and how great Meg is ✅
        Make sure this has a back button ❌ 
             [your about page has links to the other two pages, but i'm fairly certain that you need a button that takes you to the previous page you were on]

    Add a settings dialog ✅

    Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅ 
        [the gear is part of the hamburger menu and not the app bar but thats not a big issue]
    Option for dark and light mode ✅
    Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
    The above two items must be implemented with built in Veutify features ✅

    Style the main game page ✅

    Make it look like wordle.com or the Wordle mobile app ✅
    Add gradient to guesses blocks and keys (with CSS) ❌ 
         [ you do add a gradient in LetterButton, but you are not using CSS]
    Add drop shadow to keys (No CSS) ✅
    Make the app responsive (No CSS) ✅

    Include a link to your Azure App in your PR comments.✅

**Comment by alowe4** [comment]
also 
extra credit 
Add an appropriate sound when the UI letter buttons are clicked (5 points) ❌ 
[ the click sound wasn't working for me, but my machine's been having audio troubles lately ]
Add creative styling following good UI/UX practices ✅
[ looks good to me ]

File: wordle-web/src/views/AboutView.vue [hunk 2]
```diff
@@ -1,25 +1,48 @@
...
-      </v-card>
-    </v-col>
-  </v-row>
+    <v-card flat color="transparent" class="mt-16">
+      <v-card-title style="color: #76ff03" class="text-decoration-underline text-h4" align="center">
+        ChatGPT instructor review of Meg Woodford
+      </v-card-title>
+      <v-card-text class="font-italic mt-2 text-body-1">
+        "There are several reasons why Meg is a great instructor for the Eastern Washington
+        University web-development class. Firstly, Meg has extensive experience in the field of web
```
Comment (mmwoodfo) [inline wordle-web/src/views/AboutView.vue:30]:
```suggestion
        University web-development class. Firstly,  in the field of web
```
Clearly Chat GPT has not taken the class. :)

**Review by mmwoodfo — COMMENTED** [review]
This looks good, good job with the word list and keyboard colors.

**Pull request closed without merging**